client/src/search-utils.ts

```diff
@@ -32,3 +32,77 @@ export function useFocusViaKeyboard(
     };
   }, [inputRef]);
 }
+
+const COLLECTION_ITEMS_KEY: string = "collection-items";
+const COLLECTION_ITEMS_UPDATED_DATE_KEY: string =
+  "collection-items-updated-date";
+
+type Item = {
+  id: number;
+  title: string;
+  url: string;
+};
+
+function getCollectionItemsUpdatedDate(): Date | null {
+  let dateString = window?.localStorage?.getItem(
+    COLLECTION_ITEMS_UPDATED_DATE_KEY
+  );
+  if (!dateString) {
+    return null;
+  }
+  return new Date(dateString);
+}
+
+function setCollectionItemsUpdatedDate(updated: Date) {
+  window?.localStorage?.setItem(
+    COLLECTION_ITEMS_UPDATED_DATE_KEY,
+    updated.toISOString()
+  );
+}
+
+export function getCollectionItems(): Item[] {
+  return JSON.parse(
+    window?.localStorage?.getItem(COLLECTION_ITEMS_KEY) || "[]"
+  );
+}
+
+function setCollectionItems(items: Item[]) {
+  window?.localStorage?.setItem(COLLECTION_ITEMS_KEY, JSON.stringify(items));
+  window?.mdnWorker?.mutationCounter ?? window.mdnWorker.mutationCounter++;
```

Comment:
Wondering if this could be a boolean flag that is just toggled on each mutation?

Reply:
We should replace this sometime soon as we won't depend on the service worker for this. I wanted to keep this PR small.